controllers/storagecluster/kms_resources.go

```diff
@@ -39,6 +39,29 @@ var (
 	_ kmsConfigMapValidateFunc = reachKMSProvider
 )
 
+type kmsResourceManager struct{}
+
+func (kmsRMgr *kmsResourceManager) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	// if 'KMS' is not enabled, nothing to delete
+	if !sc.Spec.Encryption.KeyManagementService.Enable {
+		return nil
+	}
+	kmsConfigMap, err := getKMSConfigMap(sc, r.Client, nil)
+	if err != nil {
+		return err
+	}
+	err = r.Client.Delete(context.TODO(), kmsConfigMap)
```

Comment:
Suppose `kmsconfigMap` is still in deleting phase should we go ahead and delete the `token` if no I think we should check if `kmsconfigMap` gets deleted.

Reply:
They are independent, so yes we could delete kmsToken whether any error occurs (or not) during kmsConfigMap deletion... Will make the change...